src/domain/profile/hooks/useProfileFetcher.ts

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import { useLazyQuery } from '@apollo/client';
+import * as Sentry from '@sentry/browser';
+
+import { useIsAuthorizationReady } from '../../auth/useIsAuthorizationReady';
+import useReduxProfileState from './useReduxProfileState';
+import {
+  ProfileQuery,
+  ProfileQueryDocument,
+} from '../../api/generatedTypes/graphql';
+import type { UpdateProfileType } from '../ProfileContext';
+
+function useProfileFetcher({
+  setProfileToContext,
+}: {
+  setProfileToContext: UpdateProfileType;
+}) {
+  const [isFullyCalled, setIsFullyCalled] = React.useState(false);
+  const [isLoginReady, loginInProgress] = useIsAuthorizationReady();
+
+  // TODO: Get rid of redux store
+  const { onSuccess: onSuccessForRedux, onError: onErrorForRedux } =
+    useReduxProfileState();
+
+  const [fetchProfile, { loading, refetch, called }] =
+    useLazyQuery<ProfileQuery>(ProfileQueryDocument, {
+      onCompleted: (data) => {
+        setProfileToContext(data?.myProfile || null);
+        onSuccessForRedux(data);
+        // eslint-disable-next-line no-console
+        console.info('Guardian profile fetched and set in the context.');
+        setIsFullyCalled(true);
+      },
+      onError: (error) => {
+        // eslint-disable-next-line no-console
+        console.error(error);
+        setProfileToContext(null);
+        setIsFullyCalled(false);
+        onErrorForRedux(error);
+        Sentry.captureException(error);
+      },
+    });
+
+  const isLoading = Boolean(loginInProgress || loading);
+
+  // Fetch the profile information from the Kukkuu API
+  // when the login provider has authenticated successfully to serve the API tokens.
+  React.useEffect(() => {
+    if (!called && isLoginReady) {
+      // eslint-disable-next-line no-console
+      console.info('Fetching profile from Kukkuu API...');
+      fetchProfile();
+    }
+    return () => {
+      setIsFullyCalled(false);
+      // eslint-disable-next-line no-console
+      console.info('Cleaned the fetchProfile effect.');
+    };
+  }, [called, fetchProfile, isLoginReady]);
```

Comment:
A bit concerned whether the isFullyCalled setting to true/false is working correctly here between the lazy query and the useEffect cleaner. I mean the useEffect's cleaner sets isFullyCalled to false but onCompleted in the lazy query sets it to true. Is there any possibility that onCompleted gets run and then the useEffect's cleaner gets run after that which would make the isFullyCalled go from false->true->false? Please would you check how this works before merging, but I'll give you the benefit of the doubt here and approve this with the assumption that you consider these comments and take action if you feel necessary. Thank you!

Reply:
How I understand it, what it does, is that if something would break the fetchProfile executing, the state would be reset to false.

But, I can retest this a bit and check how ofthen the cleanup is called and whether the cleanup is needed at all.

Reply:
To me it seems to work OK. This is a screenshot of the info messages when I navigated through the app many times (back and forth):
<img width="694" alt="image" src="https://github.com/City-of-Helsinki/kukkuu-ui/assets/389204/083369b8-53a4-4d1a-a8ef-2193cf08be18">

Also the reload of the page works OK:
<img width="701" alt="image" src="https://github.com/City-of-Helsinki/kukkuu-ui/assets/389204/dd3566cf-4227-4535-8601-49623522d7b2">

What happens is that the reload will unmount the hook, which will then call the cleanup-function. The fetching is called too when the hook is remounted, so then the status is set back. The cleanup function might be needless, because the initial value is `false` too.

Reply:
Lets test it how it works!